### PR TITLE
Add the thanos-receive-controller with a catch all tenancy

### PR DESCRIPTION
**Pull request by metalmatze**
Add the thanos-receive-controller to operator the current thanos-receive statefulset with a catch-all tenancy.

/cc @squat @kakkoyun @aditya-konarde @brancz

### CODE REVIEW

File: tenants.libsonnet [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[
+  {
+    hashring: 'thanos-receive',
```
Comment (squat) [inline tenants.libsonnet]: let's give this hashring a more descriptive name as every hashring will be made of `thanos-receive`. maybe `default` as it is the catch-all

**Comment by metalmatze** [comment]
Pushed all the updates including renaming the statefulset to `thanos-receive-default`, add ServiceAccount, Role and RoleBinding and then update to Thanos `v0.6.0-rc.0` as default version.

/cc @squat

File: components/thanos-receive-controller.libsonnet [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        ) +
+        service.mixin.metadata.withNamespace('observatorium'),
+
+      deployment:
+        local deployment = k.apps.v1.deployment;
+        local container = deployment.mixin.spec.template.spec.containersType;
+        local env = container.envType;
+
+        local c =
+          container.new($.thanos.receiveController.deployment.metadata.name, 'quay.io/observatorium/thanos-receive-controller:latest') +
```
Comment (bwplotka) [inline components/thanos-receive-controller.libsonnet]: Can we pin to any version at some point?

Comment (squat) [reply components/thanos-receive-controller.libsonnet]: yes would be nice  to add proper master+hash tagging to the build system like thanos has

File: environments/kubernetes/kube-thanos.libsonnet [hunk 3]
```diff
@@ -38,7 +43,11 @@ local deployment = k.apps.v1.deployment;
         service.mixin.metadata.withNamespace(namespace),
       statefulSet+: {
         metadata+: {
+          name: 'thanos-receive-default',
```
Comment (bwplotka) [inline environments/kubernetes/kube-thanos.libsonnet]: I think making store `sts.mixin.spec.withReplicas(3),` is not that helpful as you have headless service with `dnssrv` option. This means that you treat each store as separate source that will be touch 100% time. If request will bring it down all would go down. However it increase availability on other crashes / restarts (: 

We explain this more: https://github.com/improbable-eng/thanos/issues/1083

File: environments/kubernetes/kube-thanos.libsonnet [hunk 4]
```diff
@@ -38,7 +43,11 @@ local deployment = k.apps.v1.deployment;
...
           namespace: namespace,
+          labels+: {
+            'controller.receive.thanos.io': 'thanos-receive-controller',
```
Comment (bwplotka) [inline environments/kubernetes/kube-thanos.libsonnet]: Might be worth to configure it explicitly

**Comment by metalmatze** [comment]
Thanks for the review @bwplotka. I've created some issues to track those comments.

File: components/thanos-receive-controller.libsonnet (hunk 2 above)
Comment (bwplotka) [reply components/thanos-receive-controller.libsonnet]: Feel free to copy the logic for that from Makefile + .circleci (:

**Pull request merged**